src/frontends/pytorch/src/op/nan_to_num.cpp

```diff
@@ -0,0 +1,55 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/frontend/pytorch/node_context.hpp"
+#include "openvino/op/convert.hpp"
+#include "openvino/op/maximum.hpp"
+#include "openvino/op/minimum.hpp"
+#include "openvino/op/select.hpp"
+#include "openvino/op/equal.hpp"
+#include "openvino/op/is_nan.hpp"
+#include "openvino/op/is_inf.hpp"
+#include "openvino/op/constant.hpp"
+
+namespace ov {
+namespace frontend {
+namespace pytorch {
+namespace op {
+
+using namespace ov::op;
+
+OutputVector translate_nan_to_num_fx(const NodeContext& context) {
+    num_inputs_check(context, 1, 4);
+    
+    auto x = context.get_input(0);
+    
```

Comment:
please check if this operation work for integer input tensor for which you just need to return this tensor as-is.
If this is of floating-point number, you also need to align types of `input`, `nan_repl`, `posinf_repl` etc.